src/views/ProjectView/ProjectView.tsx

```diff
@@ -32,28 +40,43 @@ const ProjectView = () => {
 
   useEffect(() => {
     if (projectId) {
+      dispatch(setMode('edit'));
       dispatch(setLoading({ text: 'Loading project', id: LOADING_PROJECT }));
       dispatch(getProjectThunk(projectId))
         .then((res) => res.type.includes('rejected') && navigate('/not-found'))
         .catch(Promise.reject)
         .finally(() => dispatch(clearLoading(LOADING_PROJECT)));
-    } else {
+    } else if (projectMode !== 'new') {
       navigate('/planning');
     }
-  }, [projectId]);
+    if (projectMode === 'new') {
+      dispatch(resetProject());
+    }
+
+    // if !projectId and newProject
+  }, [projectId, projectMode, navigate, dispatch]);
 
-  const navItems: Array<INavigationItem> = [
-    { route: 'basics', label: t('basicInfo'), component: <ProjectBasics /> },
-    { route: 'notes', label: t('notes'), component: <ProjectNotes /> },
-  ];
+  const getNavItems = useCallback(() => {
```

Comment:
Maybe this doesn't need to be a function, try:

  const navItems: Array<INavigationItem> = [
    { route: projectMode === 'new' ? 'new' : 'basics', label: t('basicInfo'), component: <ProjectBasics /> },
    projectMode !== 'new' && { route: 'notes', label: t('notes'), component: <ProjectNotes /> },
    ...
  ];

Reply:
I get this error by doing this

`Type 'false | { route: string; label: string; component: Element; }' is not assignable to type 'INavigationItem'.
  Type 'boolean' is not assignable to type 'INavigationItem'.`